### PR TITLE
feat: expose get dnskey and listdnskeys.

### PR DESCRIPTION
**Pull request by pmakani**

### CODE REVIEW

**Review by rahulKQL — APPROVED** [review]

File: src/main/java/com/google/cloud/dns/spi/v1/HttpDnsRpc.java [hunk 1]
```diff
@@ -380,6 +382,43 @@ public Change getChangeRequest(String zoneName, String changeRequestId, Map<Opti
...
+      DnsException serviceException = translate(ex, true);
+      if (serviceException.getCode() == HTTP_NOT_FOUND) {
+        return null;
+      }
+      throw serviceException;
+    }
+  }
+
+  @Override
+  public ListResult<DnsKey> listDnsKeys(String zoneName, Map<Option, ?> options) {
```
Comment (athakor) [inline src/main/java/com/google/cloud/dns/spi/v1/HttpDnsRpc.java:402]: add unit test for `listDnsKeys`

File: src/main/java/com/google/cloud/dns/spi/v1/HttpDnsRpc.java [hunk 2]
```diff
@@ -380,6 +382,43 @@ public Change getChangeRequest(String zoneName, String changeRequestId, Map<Opti
     return request;
   }
 
+  @Override
+  public DnsKey getDnsKey(String zoneName, String dnsKeyId, Map<Option, ?> options) {
```
Comment (athakor) [inline src/main/java/com/google/cloud/dns/spi/v1/HttpDnsRpc.java:386]: is it possible to add unit test for ` getDnsKey` ?

File: src/main/java/com/google/cloud/dns/DnsKeyInfo.java [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+    private List<DnsKeyDigest> digests;
+    private String id;
+    private Boolean isActive;
+    private Long keyLength;
+    private Integer keyTag;
+    private String kind;
+    private String publicKey;
+    private String type;
+
+    Builder setAlgorithm(String algorithm) {
```
Comment (athakor) [inline src/main/java/com/google/cloud/dns/DnsKeyInfo.java:161]: i think this method would be public

File: src/main/java/com/google/cloud/dns/DnsKeyInfo.java [hunk 4]
```diff
@@ -0,0 +1,377 @@
...
+      this.algorithm = algorithm;
+      return this;
+    }
+
+    Builder setCreationTime(String creationTime) {
```
Comment (athakor) [inline src/main/java/com/google/cloud/dns/DnsKeyInfo.java:166]: same here and below all are

Comment (pmakani) [reply src/main/java/com/google/cloud/dns/DnsKeyInfo.java:166]: these are setter read only so user don't need to set the value client side

File: src/main/java/com/google/cloud/dns/DnsKeyInfo.java (hunk 3 above)
Comment (pmakani) [reply src/main/java/com/google/cloud/dns/DnsKeyInfo.java:161]: read only

File: src/main/java/com/google/cloud/dns/spi/v1/HttpDnsRpc.java (hunk 2 above)
Comment (pmakani) [reply src/main/java/com/google/cloud/dns/spi/v1/HttpDnsRpc.java:386]: yes adding

File: src/main/java/com/google/cloud/dns/spi/v1/HttpDnsRpc.java (hunk 1 above)
Comment (pmakani) [reply src/main/java/com/google/cloud/dns/spi/v1/HttpDnsRpc.java:402]: adding

Comment (pmakani) [reply src/main/java/com/google/cloud/dns/spi/v1/HttpDnsRpc.java:402]: Done

File: src/main/java/com/google/cloud/dns/spi/v1/HttpDnsRpc.java (hunk 2 above)
Comment (pmakani) [reply src/main/java/com/google/cloud/dns/spi/v1/HttpDnsRpc.java:386]: Done

**Comment by pmakani** [comment]
@athakor PTAL.

**Review by athakor — APPROVED** [review]

**Pull request closed without merging**